src/queues.h

```diff
@@ -179,6 +179,10 @@ gint64 queues_get_next_datachange(gint64 time);
  */
 struct notification* queues_get_by_id(int id);
 
+/**
+ * Reapply all rules to the queue (used when reloading configs)
```

Comment:
What if a property of a notification is previously changed by a rule? Then reapplying the rule will apply to the changed notification, not the original. This might result in weird behavior. To make it work correctly you have to store the original notification's contents

Reply:
That's true, but for now I think this is the correct behavior. I also added it to the documentation. Maybe we could add an option `keep_original` or something to maintain a copy of the original notification if the user so wants. Doing so would add to the memory footprint

Reply:
I would definitely not make an option for this. The behavior will probably be acceptable in most cases, but the more correct thing to do is store the original notification and apply the rule to that.

Say a rule matches a notification with a timeout of 10 seconds with match_dbus_timeout and sets the timeout to 20 seconds and the color to yellow. Let's say you change your config to change the color to green instead and reload. You expect the notification to become green, but it stays yellow, because it's timeout has been changed to 20, making it not match the rule anymore.

Of course this is not the end of the world, but it is not the correct behavior. I don't see many rules that would make it a very common occurrence, so it would be fine by me to keep it like this. But it would be good to document and make an issue about it, so it is not forgotten.

Reply:
@fwsmit I would prefer doing that in another pr if possible since this is already quite big. How are the current changes looking?